src/main/java/org/opensearch/agent/tools/CreateAlertTool.java

```diff
@@ -0,0 +1,282 @@
+                ModelTensorOutput modelTensorOutput = (ModelTensorOutput) r.getOutput();
+                Map<String, ?> dataMap = Optional
+                    .ofNullable(modelTensorOutput.getMlModelOutputs())
+                    .flatMap(outputs -> outputs.stream().findFirst())
+                    .flatMap(modelTensors -> modelTensors.getMlModelTensors().stream().findFirst())
+                    .map(ModelTensor::getDataAsMap)
+                    .orElse(null);
+                if (dataMap == null) {
+                    throw new IllegalArgumentException("No dataMap returned from LLM.");
+                }
+                String response = "";
+                if (dataMap.containsKey("response")) {
+                    response = (String) dataMap.get("response");
+                    Pattern jsonPattern = Pattern.compile("```json(.*?)```", Pattern.DOTALL);
+                    Matcher jsonBlockMatcher = jsonPattern.matcher(response);
+                    if (jsonBlockMatcher.find()) {
+                        response = jsonBlockMatcher.group(1);
+                        response = response.replace("\\\"", "\"");
+                    }
+                } else {
+                    // LLM sometimes returns the tensor results as a json object directly instead of
+                    // string response, and the json object is stored as a map.
+                    response = StringUtils.toJson(dataMap);
+                }
+                if (!isJson(response)) {
+                    throw new IllegalArgumentException(
+                        LoggerMessageFormat.format(null, "The response from LLM is not a json: [{}]", response)
+                    );
+                }
+                listener.onResponse((T) response);
+            }, e -> {
+                log.error("Failed to run model " + modelId, e);
+                listener.onFailure(e);
+            }));
+        } catch (Exception e) {
+            log.error("Failed to call CreateAlertTool", e);
+            listener.onFailure(e);
+        }
+    }
+
+    private String getIndexMappingInfo(Map<String, String> parameters) throws InterruptedException, ExecutionException {
+        if (!parameters.containsKey("indices") || parameters.get("indices").isEmpty()) {
+            throw new IllegalArgumentException(
+                "No indices in the input parameter. Ask user to "
+                    + "provide index as your final answer directly without using any other tools"
+            );
+        }
+        String rawIndex = parameters.getOrDefault("indices", "");
+        List<String> indexList;
+        try {
+            indexList = gson.fromJson(rawIndex, new TypeToken<List<String>>() {
+            }.getType());
+        } catch (Exception e) {
+            // LLM sometimes returns the indices as a string instead of a json list, although we require that in the tool description.
+            indexList = Collections.singletonList(rawIndex);
```

Comment:
Is rawIndex a comma separated string? What data structure are expected to stored in the `indexLIst`: ["hello, world"] or ["hello"], ["world"]?

Reply:
We expects rawIndex to be json string like `[index_name1]` mostly or `[index_name1, index_name2]` if there are 2 indices. Just as I tell LLM in the tool description `"1. indices: The input indices of the monitor, should be a list of string in json format`. So we can parse it to indexList via gson directly.

But there is one time LLM returns indices without `[]` which is only `index_name1`, so I catch the parse failing exception and use rawIndex directly here.

Reply:
There is a UT in CreateAlertToolTests::testTool to cover this corner case

Reply:
Is it possible that LLM returns "index1, index2" ? I suggest do a string split with comma here to cover this case as well.

Reply:
I haven't see this corner case so far. Will cover this case as well